src/FileUpload/FileUploadFactory.php

```diff
@@ -40,6 +47,7 @@ public function __construct(
         $this->pathresolver = $pathresolver;
         $this->filesystem = $filesystem;
         $this->validators = $validators;
+        $this->fileNameGenerator = new FileNameGenerator\Simple();
```

Comment:
Can this be added to the constructor with it defaulting to this generator? I don't think it is fair that we
silently create objects for the user - I understand there is a `withFileNameGenerator` method.

I haven't written PHP in a pretty long time so it's fine if this might not be ___doable___ without breaking backwards compatibility

Reply:
Fixed